test-data/unit/check-unreachable-code.test

```diff
@@ -1416,3 +1416,15 @@ x = 1  # E: Statement is unreachable
 raise Exception
 x = 1  # E: Statement is unreachable
 [builtins fixtures/exception.pyi]
+
+[case testUnreachableNoReturnBinaryOps]
+# flags: --warn-unreachable
+from typing import NoReturn
+
+a: NoReturn
+a and 1  # E: Right operand of "and" is never evaluated
+a or 1  # E: Right operand of "or" is never evaluated
+a or a  # E: Right operand of "or" is never evaluated
```

Comment:
Why `Right`? Maybe `Left`?

Reply:
Right is correct, the left operand is always evaluated.